### PR TITLE
fix(Tags filter): Filter assets by tag ID

### PR DESCRIPTION
**Pull request by Vitor-Avila**
### SUMMARY
https://github.com/apache/superset/pull/28570 updated `SelectTagsValue` and `tagToSelectOption`. These were also used by the Tags filter in the CRUD views (dashboards, charts and saved queries), so these filters stopped working (as they're now configured to use the `tag.id` as the value (rather than the `tag.name` which is expected by the API filter).

This PR introduces a new filter operator for these assets to be able to filter tags by ID. I decided not to update the existing filter (that filters by name) and instead create a new one, to avoid a breaking change in the API.

I also removed the `TAGGING_SYSTEM` FF check in the `SavedQueries` API, for consistency with the `Dashboard` (https://github.com/apache/superset/pull/23909) and `Chart` (https://github.com/apache/superset/pull/24060) APIs.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

**Before**

https://github.com/apache/superset/assets/96086495/3c4c4694-210d-4fe5-852f-f2ed8f7e56e3

**After**

https://github.com/apache/superset/assets/96086495/87ff01e3-5abe-42cc-8761-f813be73e174

### TESTING INSTRUCTIONS
Added integration tests for both API filters.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [X] Required feature flags: `TAGGING_SYSTEM`
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base_api.py [hunk 1]
```diff
@@ -180,8 +180,19 @@ class BaseTagFilter(BaseFilter):  # pylint: disable=too-few-public-methods
     """ The Tag class_name to user """
     model: type[Dashboard | Slice | SqllabQuery | SqlaTable] = Dashboard
     """ The SQLAlchemy model """
+    id_based_filter = False
 
     def apply(self, query: Query, value: Any) -> Query:
+        # ID based filter
+        if self.id_based_filter:
```
Comment (hughhhh) [inline superset/views/base_api.py:187]: Can we write a small unit test to verify the query is properly grabbing the write rows?

Comment (Vitor-Avila) [reply superset/views/base_api.py:187]: Hey @hughhhh -- just added unit tests for the `BaseTagFilter` -- the `ChartTagNameFilter`, `ChartTagIdFilter`, `DashboardTagNameFilter`, `DashboardTagIdFilter`, `SavedQueryTagNameFilter` and `SavedQueryTagIdFilter` should have integration tests already 🙌 

Let me know if you have any other concerns or feedback 🙏

**Review by betodealmeida — APPROVED** [review]
Nice! I left a recommendation about splitting the `BaseTagFilter` class into two, if you think it's not a lot of additional work — it might make the code easier to follow.

File: superset/views/base_api.py [hunk 2]
```diff
@@ -180,8 +180,19 @@ class BaseTagFilter(BaseFilter):  # pylint: disable=too-few-public-methods
     """ The Tag class_name to user """
     model: type[Dashboard | Slice | SqllabQuery | SqlaTable] = Dashboard
     """ The SQLAlchemy model """
+    id_based_filter = False
```
[the remark this answers is not in the document]
Comment (Vitor-Avila) [reply superset/views/base_api.py]: that makes sense! I just pushed a new commit with this change. I also ended up moving these two filters to `superset.tag.filters` as I believe it makes more sense -- let me know your thoughts 🙏

**Pull request merged**